samples/dotnet/EdgeX509AuthDownstreamDevice/Program.cs

```diff
@@ -152,27 +153,47 @@ internal static (X509Certificate2, IEnumerable<X509Certificate2>) ParseCertifica
             return ParseCertificateAndKey(cert, privateKey);
         }
 
-        static TransportType GetTransport(string protocol)
+        static ITransportSettings[] GetTransport(string protocol, X509Certificate2 trustedCACert)
         {
-            TransportType result = TransportType.Mqtt_Tcp_Only;
+            TransportType transportType = TransportType.Mqtt_Tcp_Only;
+            ITransportSettings[] transportSettings = new ITransportSettings[1];
 
-            if (!string.IsNullOrWhiteSpace(protocol))
+            if (string.Compare("Mqtt", protocol, StringComparison.OrdinalIgnoreCase) == 0)
             {
-                if (string.Compare("MqttWs", protocol, StringComparison.OrdinalIgnoreCase) == 0)
-                {
-                    result = TransportType.Mqtt_WebSocket_Only;
-                }
-                else if (string.Compare("Amqp", protocol, StringComparison.OrdinalIgnoreCase) == 0)
-                {
-                    result = TransportType.Amqp_Tcp_Only;
-                }
-                else if (string.Compare("AmqpWs", protocol, StringComparison.OrdinalIgnoreCase) == 0)
-                {
-                    result = TransportType.Amqp_WebSocket_Only;
-                }
+                transportType = TransportType.Mqtt_WebSocket_Only;
+            }
+            else if (string.Compare("MqttWs", protocol, StringComparison.OrdinalIgnoreCase) == 0)
+            {
+                transportType = TransportType.Mqtt_WebSocket_Only;
+            }
+            else if (string.Compare("Amqp", protocol, StringComparison.OrdinalIgnoreCase) == 0)
+            {
+                transportType = TransportType.Amqp_Tcp_Only;
+            }
+            else if (string.Compare("AmqpWs", protocol, StringComparison.OrdinalIgnoreCase) == 0)
+            {
+                transportType = TransportType.Amqp_WebSocket_Only;
+            }
+            else
+            {
+                throw new ArgumentException("Invalid protocol");
             }
 
-            return result;
+            if (string.Compare("Amqp", protocol, StringComparison.OrdinalIgnoreCase) == 0)
+            {
+                transportSettings[0] = new AmqpTransportSettings(transportType);
+                AmqpTransportSettings amqpTransportSettings = (AmqpTransportSettings)transportSettings[0];
```

Comment:
Why not use the certificate callback this for all protocols. This makes things very consistent.

Reply:
My logic here was to only touch the control flow where the problem occurred to reduce potential unforeseen side effects.  You make a great point about consistency though. I will change.